src/util/invariant.h

```diff
@@ -201,16 +249,48 @@ invariant_violated_string(
 #define __this_function__ __func__
 #endif
 
-#define INVARIANT(CONDITION, REASON)                                           \
+#define EXPAND(x) x
+
+#define GET_MACRO(X1, X2, X3, X4, X5, X6, MACRO, ...) MACRO
+
+#define REDIRECT(MACRO, ...)                                                   \
```

Comment:
This is a macro dispatch mechanism.

This really needs to be documented before going in.